generic/utils/ras.py

```diff
@@ -0,0 +1,229 @@
+
+import os
+import commands
+from shutil import copyfile
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class RASTools(Test):
+
+    '''
+    This test verifies below RAS tools:
+    set_poweron_time - set_poweron_time schedules the power on time
+    sys_ident - sys_ident provides unique system identification information
+    lsmcode - lsmcode provides FW version information
+    drmgr - drmgr can be used for pci, cpu or memory hotplug
+    lsprop - lsprop provides device tree information
+    lsslot - lsslot lists the slots based on the option provided
+    lsvio - lsvio lists the virtual I/O adopters and devices
+    nvram - nvram command retrieves and displays NVRAM data
+    ofpathname - ofpathname translates the device name between logical name and Open Firmware name
+    ppc64_cpu - ppc64_cpu is used to set cpu options
+    rtas_ibm_get_vpd - rtas_ibm_get_vpd gives vpd data
+    rtas_errd and rtas_dump - rtas_errd adds RTAS events to /var/log/platform and rtas_dump dumps RTAS events
+    '''
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        global is_fail
+        print "Output of command \"%s\":" % cmd
+        cmd_result = process.run(cmd, ignore_status=True, sudo=True)
+        if cmd_result.exit_status != 0:
+            self.is_fail = 1
+        return
+
+    def setUp(self):
+        architecture = os.uname()[4]
+        if "ppc" not in architecture:
+            self.skip("supported only on Power platform")
+        sm = SoftwareManager()
+        for package in ("ppc64-diag", "powerpc-utils", "lsvpd"):
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(
+                    "Fail to install %s required for this test." % package)
+        log_dir = os.path.join(self.outputdir, 'log')
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
+
+    def test1_set_poweron_time(self):
+        print "===============Executing set_poweron_time tool test==============="
+        self.run_cmd("set_poweron_time -m")
+        self.run_cmd("set_poweron_time -h")
+        self.run_cmd("set_poweron_time -d m2")
+        self.run_cmd("set_poweron_time -t M6D15h12")
+        if self.is_fail == 1:
+            self.fail("RAS check failed")
+
+    def test2_sys_ident_tool(self):
+        print "===============Executing sys_ident_tool test==============="
+        self.run_cmd("sys_ident -p")
+        self.run_cmd("sys_ident -s")
+        if self.is_fail == 1:
+            self.fail("RAS check failed")
+
+    def test3_lsmcode(self):
+        print "===============Executing lsmcode tool test==============="
+        self.run_cmd("lsmcode")
+        self.run_cmd("lsmcode -A")
+        self.run_cmd("lsmcode -v")
+        self.run_cmd("lsmcode -D")
+        path_db = commands.getoutput("sudo find / -iname vpd.db | head -1")
```

Comment:
Btw can I ask what is the purpose of this? I don't know the `vpd.db` but scanning the whole machine to try to find any matching file seems really dangerous.

Also the `head -1` would likely report the first error message so you should use `2>/dev/null` in the `find` command, but please reconsider the whole find...